avocado/core/safeloader.py

```diff
@@ -111,12 +111,21 @@ def add_imported_object(self, statement):
         if getattr(statement, 'module', None) is not None:
             module_path = statement.module.replace('.', os.path.sep)
             path = os.path.join(path, module_path)
+        else:
+            # Module has no name, its path is relative to the directory
+            # structure
+            level = getattr(statement, 'level', 0)
+            for _ in range(level - 1):
+                path = os.path.dirname(path)
         for name in statement.names:
             path = os.path.join(path, name.name.replace('.', os.path.sep))
-            if name.asname is None:
-                self.imported_objects[name.name] = path
-            else:
-                self.imported_objects[name.asname] = path
+            final_name = self._get_name_from_alias_statement(name)
+            self.imported_objects[final_name] = path
+
+    @staticmethod
+    def _get_name_from_alias_statement(alias):
+        '''Returns the aliased name or original one.'''
```

Comment:
Double quotes here.

Reply:
Thanks for the review, I'll send a new version with s/'/"/g.